app/src/main/java/com/example/studentarena/fragments/MessageFragment.java

```diff
@@ -2,65 +2,131 @@
+import java.util.Set;
 
-    // TODO: Rename and change types of parameters
-    private String mParam1;
-    private String mParam2;
+public class MessageFragment extends Fragment {
+    private static final String TAG = "Message Fragment";
+    RecyclerView rvMessages;
+    protected MessageAdapter adapter;
+    private User currentUser;
+    private List<Message> messageList;
 
     public MessageFragment() {
         // Required empty public constructor
     }
 
-    /**
-     * Use this factory method to create a new instance of
-     * this fragment using the provided parameters.
-     *
-     * @param param1 Parameter 1.
-     * @param param2 Parameter 2.
-     * @return A new instance of fragment MessageFragment.
-     */
-    // TODO: Rename and change types and number of parameters
-    public static MessageFragment newInstance(String param1, String param2) {
-        MessageFragment fragment = new MessageFragment();
-        Bundle args = new Bundle();
-        args.putString(ARG_PARAM1, param1);
-        args.putString(ARG_PARAM2, param2);
-        fragment.setArguments(args);
-        return fragment;
-    }
-
-    @Override
-    public void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-        if (getArguments() != null) {
-            mParam1 = getArguments().getString(ARG_PARAM1);
-            mParam2 = getArguments().getString(ARG_PARAM2);
-        }
-    }
-
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         // Inflate the layout for this fragment
         return inflater.inflate(R.layout.fragment_message, container, false);
     }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        messageList = new ArrayList<>();
+        adapter = new MessageAdapter(messageList,getContext());
+        rvMessages = view.findViewById(R.id.rvMessages);
+        rvMessages.setAdapter(adapter);
+        rvMessages.setHasFixedSize(true);
+        rvMessages.setLayoutManager(new LinearLayoutManager(getContext()));
+        currentUser = (User) ParseUser.getCurrentUser();
+        getMessage();
+    }
+
+    private void getMessage() {
+        ParseQuery<Message> query1 = ParseQuery.getQuery(Message.class);
+        query1.whereEqualTo(Message.SENDER_KEY, (User) ParseUser.getCurrentUser());
+
+        ParseQuery<Message> query2 = ParseQuery.getQuery(Message.class);
+        query2.whereEqualTo(Message.RECEIVER_KEY, (User) ParseUser.getCurrentUser());
+
+        List<ParseQuery<Message>> list = new ArrayList<ParseQuery<Message>>();
+        list.add(query1);
+        list.add(query2);
+
+        ParseQuery<Message> query = ParseQuery.or(list);
+        query.include(Message.POST_KEY);
+        MessageServices newMessageService = new MessageServices(new MessageInterface() {
```

Comment:
i don't know how this service stuff works, pls explain to me tomorrow. i told you, you know android development better than i do :)